src/components/DeliverableView/QuestionsDeliverableEditForm.tsx

```diff
@@ -181,12 +188,41 @@ const QuestionsDeliverableEditForm = (props: QuestionsDeliverableEditViewProps):
     }
   }, [exit, updateSuccess, uploadSuccess]);
 
+  const validateForm = () => {
+    const allRequiredVariables = stagedVariableWithValues.filter((v) => v.isRequired);
+
+    const missingRequiredFields = allRequiredVariables.some((variable) => {
+      let hasEmptyValue = false;
+      const values = variable.values;
+      if (!values || values.length === 0 || getRawValue(variable) === undefined || getRawValue(variable) === '') {
+        hasEmptyValue = true;
+      }
+      return hasEmptyValue;
+    });
+
+    if (missingRequiredFields) {
+      return false;
+    }
+
+    return true;
+  };
+
   const handleOnSave = () => {
     if (pendingVariableValues.size === 0) {
       // If the user clicks save but there are no changes, just navigate them back to the deliverable
       exit();
     }
 
+    if (!validateForm()) {
+      setValidateFields(true);
+      return;
+    }
+
+    // If Questionnaire Deliverable is part of the Application/Pre-screen mark deliverable as “Completed”
+    if (isPrescreen) {
+      updateDeliverable({ ...deliverable, status: 'Completed' });
```

Comment:
This will fail if used by an external user.
Our BE always will run this step so I don't think we need this.

Reply:
But in the [Jira ticket](https://terraformation.atlassian.net/browse/SW-5697) Nick and @sgrimm said we needed to do this in the FE. Did I misunderstood?

Reply:
I missed those updates previously. In this case, we need to use the `completeDeliverable` hook instead.

Reply:
Which I just realized we don't have yet... So we need to add the Redux for it as well. 

It should be identical to `DeliverableService.submit()`, but just with `/{deliverableId}/submissions/{projectId}/complete` instead. 

https://github.com/terraware/terraware-web/blob/main/src/services/DeliverablesService.ts#L164

https://github.com/terraware/terraware-web/blob/main/src/redux/features/deliverables/deliverablesAsyncThunks.ts#L82

https://github.com/terraware/terraware-web/blob/main/src/redux/features/deliverables/deliverablesSlice.ts#L65

https://github.com/terraware/terraware-web/blob/main/src/components/DeliverableView/useSubmitDeliverable.ts

Reply:
Sorry this turns out to be a lot of changes required.

Reply:
No worries, and thanks for the feedback. Will update it!